alicloud/data_source_alicloud_ddoscoo_instances_test.go

```diff
@@ -0,0 +1,58 @@
+package alicloud
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/terraform-providers/terraform-provider-alicloud/alicloud/connectivity"
+
+	"github.com/hashicorp/terraform/helper/resource"
+)
+
+func TestAccAlicloudDdoscooInstanceDataSource_basic(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			testAccPreCheckWithRegions(t, true, connectivity.DdoscooSupportedRegions)
+			testAccPreCheck(t)
+		},
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccCheckAlicloudDdoscooInstanceDataSourceBasic(acctest.RandInt()),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAlicloudDataSourceID("data.alicloud_ddoscoo_instances.instance"),
+					resource.TestCheckResourceAttr("data.alicloud_ddoscoo_instances.instance", "instances.#", "1"),
+					resource.TestCheckResourceAttrSet("data.alicloud_ddoscoo_instances.instance", "instances.0.id"),
+					resource.TestCheckResourceAttr("data.alicloud_ddoscoo_instances.instance", "instances.0.bandwidth", "30"),
+					resource.TestCheckResourceAttr("data.alicloud_ddoscoo_instances.instance", "instances.0.base_bandwidth", "30"),
+					resource.TestCheckResourceAttr("data.alicloud_ddoscoo_instances.instance", "instances.0.service_bandwidth", "100"),
+					resource.TestCheckResourceAttr("data.alicloud_ddoscoo_instances.instance", "instances.0.port_count", "50"),
+					resource.TestCheckResourceAttr("data.alicloud_ddoscoo_instances.instance", "instances.0.domain_count", "50"),
+				),
```

Comment:
please add another step that check the instance is not exist by setting a fake name_regex. You can refer to dns datasource testcases.